tm2/pkg/std/tx.go

```diff
@@ -107,14 +107,13 @@ func (tx Tx) GetMemo() string { return tx.Memo }
 // .Empty().
 func (tx Tx) GetSignatures() []Signature { return tx.Signatures }
 
-func (tx Tx) GetSignBytes(chainID string, accountNumber uint64, sequence uint64) ([]byte, error) {
+func (tx Tx) GetSignBytes(chainID string, sequence uint64) ([]byte, error) {
 	return GetSignaturePayload(SignDoc{
-		ChainID:       chainID,
-		AccountNumber: accountNumber,
```

Comment:
I'm considering whether it makes sense to replace AccountNumber with the pubkey here. However, since we're signing with the corresponding privkey and appending the pubkey to the tx, it may be unnecessary. 

cc @aeddi @kristovatlas

Reply:
I still lack some context and a deep knowledge of the codebase, but after some live discussions, reading the PR on Cosmos, and considering the arguments presented in the other comments, I feel that it's safe to remove this value. In fact, if it is truly unnecessary, it is important to eliminate it for the sake of clarity, especially when it comes to data related to security.

About adding a pub key field, there is no point in doing it IMO.